public/app/features/templating/DefaultTemplateQueryComponent.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { PureComponent } from 'react';
+import { TemplateQueryProps } from 'app/types/plugins';
+
+export default class DefaultTemplateQueryComponent extends PureComponent<TemplateQueryProps, any> {
+  constructor(props) {
+    super(props);
+    this.state = { value: props.query };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleBlur = this.handleBlur.bind(this);
```

Comment:
We've used the "on" prefix and written these as arrow functions. That way we don't have to bind them in the constructor.

Reply:
Fixed